hazelcast/src/test/java/com/hazelcast/partition/impl/InternalPartitionServiceImplTest.java

```diff
@@ -27,16 +29,106 @@
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
 
 @RunWith(HazelcastParallelClassRunner.class)
 @Category({QuickTest.class, ParallelTest.class})
 public class InternalPartitionServiceImplTest extends HazelcastTestSupport {
 
+    private HazelcastInstance instance = createHazelcastInstance();
+    private InternalPartitionService partitionService = getPartitionService(instance);
+
     @Test(expected = HazelcastInstanceNotActiveException.class)
     public void test_getPartitionOwnerOrWait_throwsException_afterNodeShutdown() throws Exception {
-        HazelcastInstance node = createHazelcastInstance();
-        InternalPartitionService partitionService = getPartitionService(node);
-        node.shutdown();
+        instance.shutdown();
+        partitionService.getPartitionOwnerOrWait(0);
+    }
+
+    @Test
+    public void test_initialAssignment() {
+        partitionService.firstArrangement();
+
+        int partitionCount = partitionService.getPartitionCount();
+        for (int i = 0; i < partitionCount; i++) {
+            assertTrue(partitionService.isPartitionOwner(i));
+        }
+    }
+
+    @Test
+    public void test_initialAssignment_whenClusterNotActive() {
+        instance.getCluster().changeClusterState(ClusterState.FROZEN);
+
+        partitionService.firstArrangement();
+        assertNull(partitionService.getPartitionOwner(0));
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void test_getPartitionOwnerOrWait_whenClusterNotActive() {
+        instance.getCluster().changeClusterState(ClusterState.FROZEN);
+
+        partitionService.firstArrangement();
         partitionService.getPartitionOwnerOrWait(0);
     }
-}
\ No newline at end of file
+
+    @Test
+    public void test_setInitialState() {
+        Address thisAddress = getNode(instance).getThisAddress();
+        int partitionCount = partitionService.getPartitionCount();
+        Address[][] addresses = new Address[partitionCount][MAX_REPLICA_COUNT];
+        for (int i = 0; i < partitionCount; i++) {
+            addresses[i][0] = thisAddress;
+        }
+
+        InternalPartitionServiceImpl partitionServiceImpl = (InternalPartitionServiceImpl) partitionService;
+        partitionServiceImpl.setInitialState(addresses);
+
+        for (int i = 0; i < partitionCount; i++) {
+            assertTrue(partitionService.isPartitionOwner(i));
+        }
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void test_setInitialState_multipleTimes() {
+        Address thisAddress = getNode(instance).getThisAddress();
+        int partitionCount = partitionService.getPartitionCount();
+        Address[][] addresses = new Address[partitionCount][MAX_REPLICA_COUNT];
+        for (int i = 0; i < partitionCount; i++) {
+            addresses[i][0] = thisAddress;
+        }
+
+        InternalPartitionServiceImpl partitionServiceImpl = (InternalPartitionServiceImpl) partitionService;
```

Comment:
I would do this in the setup as well. Just make a field partitionService of type InternalPartitionServiceImpl and do the casting in the setup.